src/ILCompiler.Compiler/src/Compiler/CompilationModuleGroup.cs

```diff
@@ -47,7 +51,7 @@ public void AddWellKnownTypes()
         {
             var stringType = _typeSystemContext.GetWellKnownType(WellKnownType.String);
 
-            if (IsTypeInCompilationGroup(stringType))
+            if (IsInCompilationGroup(stringType))
             {
                 _rootProvider.AddCompilationRoot(stringType, "String type is always generated");
```

Comment:
it seems that the reason on AddCompilationRoot is counter to the conditional, since the string type is not always generated (Case of not being in the compilation group).